lib/core/models/repositories/roleRepository.js

```diff
@@ -181,25 +195,118 @@ class RoleRepository extends Repository {
    * @returns Promise
    */
   validateAndSaveRole (role, options) {
-    return role.validateDefinition()
+    let errors;
+
+    return role
+      .validateDefinition()
       .then(() => {
         if (role._id === 'anonymous' && !role.canLogIn()) {
           errorsManager.throw('security', 'role', 'login_required');
         }
 
-        this.kuzzle.emit(
-          'core:roleRepository:save',
-          { _id: role._id, controllers: role.controllers });
-
-        return this.persistToDatabase(role, options);
+        this.kuzzle.emit('core:roleRepository:save', {
+          _id: role._id,
+          controllers: role.controllers
+        });
       })
+      .then(() => this.checkRoleNativeRights(role))
+      .then(err => {
+        err = this.checkRolePluginsRights(role);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "err". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)